packages/flutter_tools/lib/src/test/coverage_collector.dart

```diff
@@ -204,20 +204,17 @@ Future<Map<String, dynamic>> collect(Uri serviceUri, bool Function(String) libra
   int i = 0;
   while (i < kPollAttempts) {
     await isolate.load();
-    if (isolate.pauseEvent?.kind == ServiceEvent.kPauseStart) {
+    if (isolate.pauseEvent?.isPauseEvent == true) {
```

Comment:
I've recently discovered that this value is not reliable.  We may want to look at migrating all of this over to vm service lib if we can.

Reply:
By not reliable, I mean that I've been able to create multiple scenarios where it's not up to date with the actual state of the isolate

Reply:
Oh - that is good to know actually. We already depend on the other vmservice lib, I'll pull that in instead

Reply:
I checked in with some more manual debugging, as well as with the other vm service library. I found that the isolate still could not be paused, except by manually pausing in the observatory. The isolate's stack was empty leading me to believe that `Isolate.pause` only works if it is still in the midst of the event loop.

Reply:
@a-siva Does the Dart Isolate object keep the underlying VM objects (Isolate, message handler, etc.) resident? If so, I'd expect Isolate.pause() to update the Isolate state as reported by the VM service.